frontend/app/src/features/matching/components/SelectPreferencesCard.tsx

```diff
@@ -33,21 +34,23 @@ type PreferencesFormValues = {
     | undefined;
 };
 
-export const SelectPreferencesCard = () => {
+type Props = {
+  joinCallback: (preferences: Preferences) => void;
+};
+
+export const SelectPreferencesCard = ({joinCallback}: Props) => {
   const {
     handleSubmit,
     control,
     formState: { errors, isSubmitting },
   } = useForm<PreferencesFormValues>();
 
   const onSubmit = handleSubmit(async values => {
-    // TODO: @Joel - Add logic to join room
     const parsedValues = {
       difficulty: values.difficulty?.map(difficulty => difficulty.value),
       category: values.category?.map(category => category.value),
     };
-
-    console.log(parsedValues);
+    joinCallback(parsedValues);
```

Comment:
don't have to fix this now but do think about how error handling (e.g. what shd happen if the joining fails, etc.)